lib/fxpay/index.js

```diff
@@ -34,63 +34,72 @@
   };
 
 
-  exports.validateAppReceipt = function validateAppReceipt(callback) {
+  exports.validateAppReceipt = function validateAppReceipt() {
     settings.initialize();
     var defaultProductInfo = {};
-    utils.getAppSelf(function(error, appSelf) {
-      if (error) {
-        return callback(error, defaultProductInfo);
-      }
-      if (!appSelf) {
-        return callback(errors.PayPlatformUnavailable(
-                            'mozApps.getSelf() required for receipts'),
-                            defaultProductInfo);
-      }
-      var allAppReceipts = [];
-
-      receipts.all(function(error, allReceipts) {
+    return new Promise(function(resolve, reject) {
+      utils.getAppSelf(function(error, appSelf) {
         if (error) {
-          return callback(error, defaultProductInfo);
+          return reject({error: error, productInfo: defaultProductInfo});
         }
+        if (!appSelf) {
+          return reject({error: errors.PayPlatformUnavailable(
```

Comment:
I wonder if it would be possible to more directly pass a single error instance back in the rejection. Might be cleaner  - but I suppose it will depend on what additional information you want. But like you did here the productInfo was part of the error object not just passed through alongside the error.